src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1176,6 +1176,41 @@ def _configure_security_policy_storage_params(arg_ctx):
                    'for use with key management services like Azure KeyVault. '
                    'If identity is already assigned - do nothing.')
 
+    #####
+    #           sql managed instance key
+    #####
+    with self.argument_context('sql mi key') as c:
+        c.argument('managed_instance_name',
+                   arg_type=managed_instance_param_type)
+
+        c.argument('key_name',
+                   options_list=['--name', '-n'])
+
+        c.argument('kid',
+                   options_list=['--kid', '-k'],
```

Comment:
extract argument type (similar to `managed_instance_param_type`) so that the help isn't duplicated.